app.py

```diff
@@ -355,6 +376,31 @@ def main():
 	"""Redirect to main page."""
 	return redirect(url_for('login'))
 
+@app.route('/annotate/direct_entry', methods=['GET'])
+@loginrequired
+def direct_entry():
+	"""Directly enter a sentence."""
+	sent = request.args.get('sent', '').strip()
+	sentid = str(request.args.get('id', '')).strip()
+	if len(sent) == 0:
+		return jsonify({'error': 'Sentence is empty.'})
+	elif len(sentid) == 0:
+		return jsonify({'error': 'Sentence ID is empty.'})
+	db = getdb()
+	cur = db.execute(
+		'SELECT * FROM entries ORDER BY sentno ASC'
+	)
+	entries = cur.fetchall()
+	existing_ids = [entry[0] for entry in entries] + [entry[3] for entry in QUEUE]
```

Comment:
Why are we getting sentences from the database AND the queue? Is the queue just the unsaved ones?

Reply:
If there is potential overlap between `entries` and `QUEUE` I'd use a set here: `existing_ids = {entry[0] for entry in entries} | {entry[3] for entry in QUEUE}`